### PR TITLE
Repl/handle nulls in responses fix

### PR DESCRIPTION
**Pull request by Andy65**
Fixed the handling of null responses for open shift request syncs.

### CODE REVIEW

File: Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs [hunk 1]
```diff
@@ -415,158 +408,159 @@ internal async Task ProcessOpenShiftsRequests(string isRequestFromLogicApp)
...
-                            var entityToUpdate = await this.openShiftRequestMappingEntityProvider.GetOpenShiftRequestMappingEntityByKronosReqIdAsync(
-                                item.Id).ConfigureAwait(false);
+                        // Update the status in Azure table storage.
+                        var entityToUpdate = await this.openShiftRequestMappingEntityProvider.GetOpenShiftRequestMappingEntityByKronosReqIdAsync(
+                            item.Id).ConfigureAwait(false);
 
-                            if (entityToUpdate != null)
+                        if (entityToUpdate != null)
+                        {
+                            if (item.StatusName == ApiConstants.ApprovedStatus && entityToUpdate.ShiftsStatus == ApiConstants.Pending)
```
Comment (ajs117) [inline Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs]: prefer .equals for string comparsions

File: Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs [hunk 2]
```diff
@@ -415,158 +408,159 @@ internal async Task ProcessOpenShiftsRequests(string isRequestFromLogicApp)
...
 
-                                    // Build the request to get the open shift from Graph.
-                                    var httpClient = this.httpClientFactory.CreateClient("ShiftsAPI");
-                                    httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", allRequiredConfigurations.ShiftsAccessToken);
-                                    using (var getOpenShiftRequestMessage = new HttpRequestMessage(HttpMethod.Get, "teams/" + user.ShiftTeamId + "/schedule/openShifts/" + entityToUpdate.TeamsOpenShiftId))
+                                // Build the request to get the open shift from Graph.
+                                var httpClient = this.httpClientFactory.CreateClient("ShiftsAPI");
+                                httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", allRequiredConfigurations.ShiftsAccessToken);
+                                using (var getOpenShiftRequestMessage = new HttpRequestMessage(HttpMethod.Get, "teams/" + user.ShiftTeamId + "/schedule/openShifts/" + entityToUpdate.TeamsOpenShiftId))
```
Comment (ajs117) [inline Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs]: a constant templated string would be nicer for this url

File: Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs [hunk 3]
```diff
@@ -415,158 +408,159 @@ internal async Task ProcessOpenShiftsRequests(string isRequestFromLogicApp)
...
+                                        approvalHttpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", allRequiredConfigurations.ShiftsAccessToken);
 
-                                            // Send Passthrough header to indicate the sender of request in outbound call.
-                                            approvalHttpClient.DefaultRequestHeaders.Add("X-MS-WFMPassthrough", allRequiredConfigurations.WFIId);
-                                            using (var httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, "teams/" + user.ShiftTeamId + "/schedule/openshiftchangerequests/" + entityToUpdate.RowKey + "/approve")
+                                        // Send Passthrough header to indicate the sender of request in outbound call.
+                                        approvalHttpClient.DefaultRequestHeaders.Add("X-MS-WFMPassthrough", allRequiredConfigurations.WFIId);
+                                        using (var httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, "teams/" + user.ShiftTeamId + "/schedule/openshiftchangerequests/" + entityToUpdate.RowKey + "/approve")
```
Comment (ajs117) [inline Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs]: same here, template string would be nicer

File: Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs [hunk 4]
```diff
@@ -415,158 +408,159 @@ internal async Task ProcessOpenShiftsRequests(string isRequestFromLogicApp)
...
+                                        this.telemetryClient.TrackTrace($"{Resource.ProcessOpenShiftsRequests} - Getting Open Shift for {entityToUpdate.TeamsOpenShiftId} results in {getOpenShiftResponse.Content.ToString()}");
                                     }
                                 }
+                            }
 
-                                if ((item.StatusName == ApiConstants.Retract || item.StatusName == ApiConstants.Refused) && entityToUpdate.ShiftsStatus == ApiConstants.Pending)
-                                {
-                                    entityToUpdate.KronosStatus = item.StatusName;
-                                    entityToUpdate.ShiftsStatus = item.StatusName;
+                            if ((item.StatusName == ApiConstants.Retract || item.StatusName == ApiConstants.Refused) && entityToUpdate.ShiftsStatus == ApiConstants.Pending)
```
Comment (ajs117) [inline Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs]: same here, .equals for string comparsion

File: Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs [hunk 5]
```diff
@@ -415,158 +408,159 @@ internal async Task ProcessOpenShiftsRequests(string isRequestFromLogicApp)
...
+                                declineHttpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", allRequiredConfigurations.ShiftsAccessToken);
 
-                                    // Send Passthrough header to indicate the sender of request in outbound call.
-                                    declineHttpClient.DefaultRequestHeaders.Add("X-MS-WFMPassthrough", allRequiredConfigurations.WFIId);
-                                    using (var declineRequestMessage = new HttpRequestMessage(HttpMethod.Post, "teams/" + user.ShiftTeamId + "/schedule/openshiftchangerequests/" + entityToUpdate.RowKey + "/decline")
+                                // Send Passthrough header to indicate the sender of request in outbound call.
+                                declineHttpClient.DefaultRequestHeaders.Add("X-MS-WFMPassthrough", allRequiredConfigurations.WFIId);
+                                using (var declineRequestMessage = new HttpRequestMessage(HttpMethod.Post, "teams/" + user.ShiftTeamId + "/schedule/openshiftchangerequests/" + entityToUpdate.RowKey + "/decline")
```
Comment (ajs117) [inline Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs]: same here, template string would be nicer

**Review by ajs117 — CHANGES_REQUESTED** [review]

File: Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs (hunk 1 above)
Comment (Andy65) [reply Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs]: Done

File: Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs (hunk 2 above)
Comment (Andy65) [reply Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs]: Done

File: Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs (hunk 3 above)
Comment (Andy65) [reply Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs]: Done

File: Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs (hunk 4 above)
Comment (Andy65) [reply Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs]: Done

File: Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs (hunk 5 above)
Comment (Andy65) [reply Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/OpenShiftRequestController.cs]: Done

**Review by ajs117 — APPROVED** [review]

**Pull request merged**